wallet/src/de/schildbach/wallet/ui/dashpay/ContactSearchResultsAdapter.kt

```diff
@@ -0,0 +1,240 @@
+        }
+    }
+    override fun onBindViewHolder(holder: ViewHolder, position: Int, payloads: List<Any?>) {
+        if (payloads.isEmpty()) {
+            super.onBindViewHolder(holder, position, payloads)
+        } else {
+            holder.itemView.alpha = 1f
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return results[position].viewType
+    }
+
+    open inner class ViewHolder(resId: Int, inflater: LayoutInflater, parent: ViewGroup) :
+            RecyclerView.ViewHolder(inflater.inflate(resId, parent, false)) {
+
+        private val avatar by lazy { itemView.findViewById<ImageView>(R.id.avatar) }
+        private val username by lazy { itemView.findViewById<TextView>(R.id.username) }
+        private val displayName by lazy { itemView.findViewById<TextView>(R.id.displayName) }
+
+        open fun bind(usernameSearchResult: UsernameSearchResult) {
+            val defaultAvatar = UserAvatarPlaceholderDrawable.getDrawable(itemView.context,
+                    usernameSearchResult.username[0])
+
+            val dashPayProfile = usernameSearchResult.dashPayProfile
+            if (dashPayProfile.displayName.isEmpty()) {
+                displayName.text = dashPayProfile.username
+                username.text = ""
+            } else {
+                displayName.text = dashPayProfile.displayName
+                username.text = usernameSearchResult.username
+            }
+
+            if(dashPayProfile.avatarUrl.isNotEmpty()) {
+                Glide.with(avatar).load(dashPayProfile.avatarUrl).circleCrop()
+                        .placeholder(defaultAvatar).into(avatar)
+            } else {
+                avatar.background = defaultAvatar
+            }
+
+            itemClickListener?.let { l ->
+                this.itemView.setOnClickListener {
+                    l.onItemClicked(it, usernameSearchResult)
+                }
+            }
+        }
+    }
+
+
+    inner class ContactViewHolder(inflater: LayoutInflater, parent: ViewGroup) :
+            ViewHolder(R.layout.contact_row, inflater, parent) {
+
+        override fun bind(usernameSearchResult: UsernameSearchResult) {
+            super.bind(usernameSearchResult)
+            // background color alternates based on first letter
+            val color = if (usernameSearchResult.dashPayProfile.username[0].toLowerCase().toInt() % 2 != 0)
+                R.color.white
+            else
+                R.color.dash_lighter_gray
+            itemView.setBackgroundColor(itemView.resources.getColor(color))
+        }
+    }
+
+    inner class ContactRequestViewHolder(inflater: LayoutInflater, parent: ViewGroup) :
+            ViewHolder(R.layout.contact_request_row, inflater, parent) {
+
+        override fun bind(usernameSearchResult: UsernameSearchResult) {
+            super.bind(usernameSearchResult)
+            itemView.apply {
+                accept_contact_request.setOnClickListener {
```

Comment:
is it better to use `accept_contact_request` which is from `import kotlinx.android.synthetic.main.contact_request_row.view.*`

or something like this:
`private val avatar by lazy { itemView.findViewById<ImageView>(R.id.avatar) }` to initialize the object that points to the button.

Reply:
Great. I will leave this for now.